src/main/java/com/parsing/parsers/pdf/parsing/TableDetector.java

```diff
@@ -0,0 +1,105 @@
+        double height = Math.sqrt(Math.pow((y3 - y2), 2));
+        return new Rect((int) x1, (int) y3, (int) width, (int) height);
+    }
+
+    public boolean isTableExistOnPage(String fileSource) {
+        OpenCV.loadLocally();
+        Mat dst = new Mat(), cdst = new Mat(), cdstP, cropTable = new Mat();
+        Mat source = Imgcodecs.imread(fileSource, Imgcodecs.IMREAD_GRAYSCALE);
+        if (source.empty()) {
+            log.warn("Error opening image! Program Arguments: [image_name -- default " + fileSource + "] \n");
+            System.exit(-1);
+        }
+        Imgproc.Canny(source, dst, 50, 200, 3, false);
+        Imgcodecs.imwrite("afterCanny.png", dst);
+        Imgproc.cvtColor(dst, cdst, Imgproc.COLOR_GRAY2BGR);
+        cdstP = cdst.clone();
+        Mat linesP = new Mat();
+        Imgproc.HoughLinesP(dst, linesP, 3, Math.PI, 200, RectangleDetector.VERTICAL_LINE_LENGTH, 1);
+        log.debug("Quantity lines " + linesP.size().toString());
+        List<double[]> lines = new ArrayList<>();
+        for (int k = 0; k < linesP.rows(); k++) {
+            lines.add(linesP.get(k, 0));
+        }
+        if (lines.size() > MINIMAL_QUANTITY_LINES_FOR_TABLE) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public String detectTable(String fileSource) {
+        OpenCV.loadLocally();
+        Mat dst = new Mat(), cdst = new Mat(), cdstP, cropTable = new Mat();
+        Mat source = Imgcodecs.imread(fileSource, Imgcodecs.IMREAD_GRAYSCALE);
+        if (source.empty()) {
+            log.warn("Error opening image! Program Arguments: [image_name -- default " + fileSource + "] \n");
+            System.exit(-1);
+        }
+        Imgproc.Canny(source, dst, 50, 200, 3, false);
+        Imgcodecs.imwrite("afterCanny.png", dst);
+        Imgproc.cvtColor(dst, cdst, Imgproc.COLOR_GRAY2BGR);
+        cdstP = cdst.clone();
+        Mat linesP = new Mat();
+        Imgproc.HoughLinesP(dst, linesP, 1, Math.PI / 180, 25, 700, 10);
+        for (int x = 0; x < linesP.rows(); x++) {
+            double[] l = linesP.get(x, 0);
+            Imgproc.line(cdstP, new Point(l[0], l[1]), new Point(l[2], l[3]), new Scalar(0, 0, 255), 3, Imgproc.LINE_AA, 0);
+            if (l[1] > y1) {
+                y1 = l[1];
+                x2 = l[2];
+                y2 = l[3];
+            }
+            if (l[1] < y3) {
+                y3 = l[1];
+            }
+            if (l[2] > x2) {
+                x2 = l[2];
+            }
+            if (l[0] < x1) {
+                x1 = l[0];
+            }
+        }
+        Imgcodecs.imwrite("afterHough.png", cdstP);
+        Rect rectangle = createRect(x1, y1, x2, y2, y3);
+        try {
+            cropTable = source.submat(rectangle);
+            String fileResult = "destination.png";
+            Imgcodecs.imwrite(fileResult, cropTable);
+            return fileResult;
+        } catch (Exception ex) {
+            log.warn("can't crop image");
```

Comment:
consider providing information about image that can't be cropped.

Reply:
Done.